packages/sdk/src/handlers/contract-kit.ts

```diff
@@ -54,22 +88,33 @@ export class ContractKitTransactionHandler implements ChainHandler {
       info.action.amount.toString()
     );
 
-    this.signedTransaction = await wallet.signTransaction({
+    const txParams = await this.getTxParams(stable, gasPriceMinimum, txo);
+    return wallet.signTransaction(txParams);
+  }
+
+  protected async getNonce() {
+    return this.kit.connection.getTransactionCount(this.blockchainAddress);
+  }
+
+  protected async getTxParams(
+    stable: StableTokenWrapper,
+    gasPriceMinimum: BigNumber,
+    txo: CeloTxObject<unknown>
+  ) {
+    const nonce = this.lastNonce ? this.lastNonce++ : await this.getNonce();
+
+    return {
       to: stable.address,
       from: this.blockchainAddress,
-      gas: 100_000,
+      gas: this.gas,
```

Comment:
While I know this was there before the refactor, maybe we could use an estimation instead?

Reply:
We might use it, let's differ it to the next round